kernel/src/device/fb.rs

```
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use aster_framebuffer::{ColorMapEntry, FrameBuffer, PixelFormat, FRAMEBUFFER, MAX_CMAP_SIZE};
+use ostd::{
+    mm::{io_util::HasVmReaderWriter, HasPaddr, HasSize, VmIo},
+    Pod,
+};
+
+use crate::{
+    current_userspace,
+    events::IoEvents,
+    fs::{
+        device::{Device, DeviceId, DeviceType},
+        file_handle::Mappable,
+        inode_handle::FileIo,
+        utils::{IoctlCmd, StatusFlags},
+    },
+    prelude::*,
+    process::signal::{PollHandle, Pollable},
+};
+
+pub struct Fb;
+
+pub struct FbHandle {
+    framebuffer: Arc<FrameBuffer>,
+    offset: Mutex<usize>,
+}
+
+impl Device for Fb {
+    fn type_(&self) -> DeviceType {
+        DeviceType::Misc
+    }
+
+    fn id(&self) -> DeviceId {
+        // Same value with Linux
+        DeviceId::new(29, 0)
+    }
+
+    fn open(&self) -> Option<Result<Arc<dyn FileIo>>> {
+        let framebuffer = FRAMEBUFFER.get()?.clone();
```

This is incorrect. See my patch below. But anyway, the signature of this method will be changed in https://github.com/asterinas/asterinas/pull/2564, so this problem won't happen after that.

---

I have fixed it in the current version and will handle this carefully in future rebases.

From my perspective, here is the reason why the CI pass even with a buggy open: currently, the device file (`/dev/fb0`) of the EFI framebuffer is hardcoded with device probing. If the boot service does not supply a framebuffer, the device file will not be created. The buggy device-specific `open` will not be triggered when the framebuffer is absent because the syscall will return the error number `ENOENT`. Therefore, without the `/dev/fb0`, the CI will fail silently and still pass.